api/src/main/java/org/apache/iceberg/PartitionSpec.java

```diff
@@ -95,7 +95,7 @@ public int specId() {
   }
 
   public boolean isUnpartitioned() {
-    return fields.length < 1;
+    return fields.length < 1 || fields.stream().allMatch(f -> f.transform().getClass() == Transforms.alwaysNull().class);
```

Comment:
I'm quite new to Java, so my opinion may not align with convention. Just from a personal perspective, I don't see any harm to have all `Transform` implementation public as they share the same methods as the interface. BTW, are there any specific reasons to hide them initially (e.g. project is in the early stage and not stable so we can change them without breaking users)?

Reply:
The project is not in an early stage. We have many users running production workloads on it.

Reply:
I think we just happen to default to "private" for most things unless we have an explicit reason to open them up. Trying to prevent incompatibility with folks using the Iceberg API directly.